repos/fdbt-site/src/pages/api/deleteCap.ts

```diff
@@ -14,6 +18,30 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
         }
 
         const noc = getAndValidateNoc(req, res);
+        const cap = await getCapByNocAndId(noc, id);
+
+        if (!cap) {
+            throw new Error('unable to find cap with provided id');
+        }
+
+        const products = await getAllProductsByNoc(noc);
+        const matchingJsonLinks = products.map((product) => product.matchingJsonLink);
+        const tickets: TicketWithIds[] = await Promise.all(
+            matchingJsonLinks.map(async (link) => {
+                return await getProductsMatchingJson(link);
+            }),
+        );
+
+        const productsUsingCap = tickets.filter((ticket) => 'cap' in ticket && ticket.cap && ticket.cap.id === id);
+
+        if (productsUsingCap.length > 0) {
+            const { name } = cap.capDetails;
+            const errorMessage = `You cannot delete ${name} because it is being used in ${productsUsingCap.length} ticket(s).`;
+            const errors: ErrorInfo[] = [{ id: 'cap-0', errorMessage }];
+            updateSessionAttribute(req, VIEW_CAP, errors);
+            redirectTo(res, `/viewCaps?cannotDelete=${name}`);
```

Comment:
Why are we redirecting with a query string?

Reply:
this is the way we are doing in other delete api's

Reply:
@NitishAroraKPMG You are correct - but the other delete API's are all also incorrectly doing this. There is nothing on the /viewCaps (or /viewTimeRestrictions etc etc) pages which actually use this query string.

We need to make sure we dont blindly copy things from other parts of the repo, because they aren't always correct.